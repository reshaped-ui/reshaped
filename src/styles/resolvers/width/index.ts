import { responsiveClassNames, responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import s from "./width.module.css";

const width: T.StyleResolver<T.Width> = (value) => {
	if (!value) return {};
	const variables = responsiveVariables("--rs-w", value);
	const classNames = responsiveClassNames(
		s,
		(value) => (typeof value === "number" ? "--type-unit" : "--type-literal"),
		value,
		{ excludeValueFromClassName: true }
	);

	return { classNames: [s.root, classNames], variables };
};

export default width;
