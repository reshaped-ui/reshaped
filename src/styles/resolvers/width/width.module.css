@responsive .root[style*="--rs-w-"] {
	@variable --rs-w auto;
}

@responsive .--type {
	@value literal {
		width: var(--rs-w) !important;
	}

	@value unit {
		width: calc(var(--rs-w) * var(--rs-unit-x1)) !important;
	}
}
