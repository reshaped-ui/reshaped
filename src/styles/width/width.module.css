.literal {
	width: var(--rs-w-s, auto) !important;
}

.unit {
	width: calc(var(--rs-w-s) * var(--rs-unit-x1)) !important;
}

@each $viewport in m, l, xl {
	@media (--rs-viewport-$viewport) {
		.literal--$viewport {
			width: var(--rs-w-$viewport) !important;
		}

		.unit--$viewport {
			width: calc(var(--rs-w-$viewport) * var(--rs-unit-x1)) !important;
		}
	}
}
