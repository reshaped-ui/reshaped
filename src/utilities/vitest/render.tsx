import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
