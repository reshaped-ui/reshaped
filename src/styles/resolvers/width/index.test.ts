import { expect, test, describe } from "vitest";
import width from "./index";

describe("Styles/Width", () => {
	test("handles px value", () => {
		expect(width("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(width("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(width(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(width()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(width({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
