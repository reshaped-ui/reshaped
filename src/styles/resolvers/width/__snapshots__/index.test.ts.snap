// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Width > handles % value 1`] = `
{
  "classNames": [
    "_root_1de7fc",
    [
      "_--type-literal_1de7fc",
    ],
  ],
  "variables": {
    "--rs-w-s": "50%",
  },
}
`;

exports[`Styles/Width > handles px value 1`] = `
{
  "classNames": [
    "_root_1de7fc",
    [
      "_--type-literal_1de7fc",
    ],
  ],
  "variables": {
    "--rs-w-s": "50px",
  },
}
`;

exports[`Styles/Width > handles responsive value 1`] = `
{
  "classNames": [
    "_root_1de7fc",
    [
      "_--type-literal_1de7fc",
      "_--type-literal--l_1de7fc",
    ],
  ],
  "variables": {
    "--rs-w-l": "50%",
    "--rs-w-s": "50px",
  },
}
`;

exports[`Styles/Width > handles undefined value 1`] = `{}`;

exports[`Styles/Width > handles unit value 1`] = `
{
  "classNames": [
    "_root_1de7fc",
    [
      "_--type-unit_1de7fc",
    ],
  ],
  "variables": {
    "--rs-w-s": 5,
  },
}
`;
